### PR TITLE
Cfy 2994 create test to cover rest backwards compatiblity

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

**Comment by idanmo** [comment]
Perhaps we should put the test under test_suites/backwards so that we'll put all backwards related tests there?

File: cosmo_tester/test_suites/test_rest_api_backwards_compatibility/rest_api_backwards_compatibility_test.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            f.write('pip install cloudify=={0} >/dev/null\n'
+                    .format(cfy_client_version))
+            f.write('python {0}\n'.format(python_script_name))
+        permissions = os.stat(shell_script_name)
+        os.chmod(shell_script_name, permissions.st_mode | 0111)
+
+    def test_old_client_vs_new_server(self):
+        output = subprocess.check_output(
+            '/bin/bash {0}'.format(SHELL_SCRIPT_NAME), shell=True)
+        print('output: {0}'.format(output))
```
Comment (noakup) [inline cosmo_tester/test_suites/test_rest_api_backwards_compatibility/rest_api_backwards_compatibility_test.py]: remove the printout

File: cosmo_tester/resources/scripts/use_old_rest_client.template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            status = rest_client.manager.get_status()
+            output["exit_code"] = 0
+            output["details"] = status
+        else:
+            output["exit_code"] = 1
+            output["details"] = "rest client url is {0} instead of {1}".format(
+                rest_client_url, expected_rest_client_url)
+    except Exception as e:
+        output["exit_code"] = 1
+        output["details"] = str(e.message)
```
Comment (dankilman) [inline cosmo_tester/resources/scripts/use_old_rest_client.template]: e.message may sometimes not contain anything. it generally preferable simply doing `str(e)`

File: cosmo_tester/resources/scripts/run_old_rest_client_in_shell.template [hunk 3]
```diff
@@ -0,0 +1,8 @@
+orig_dir=$(pwd)
```
Comment (dankilman) [inline cosmo_tester/resources/scripts/run_old_rest_client_in_shell.template]: lets add `set -x` as the first line

**Pull request merged**